### ISSUE TITLE
False positives `invalid-enum-extension`

### ISSUE THREAD

**Issue by Bibo-Joshi** [pylint-dev/pylint#6887]
### Bug description

That check doesn't seem to check if the parent class actually defines enumeration members. This happens only if the parent class is defined in a differen module. MWE:

`bar.py`

```
import Enum

class StringEnum(str, Enum):
    def __repr__(self) -> str:
        return f"<{self.__class__.__name__}.{self.name}>"
```

`foo.py`

```
from bar import StringEnum


class ChatType(StringEnum):

    SENDER = "sender"
    PRIVATE = "private"
```

### Command used

```shell
`pylint foo.py`
```

### Pylint output

```shell
************* Module foo
foo.py:1:0: C0114: Missing module docstring (missing-module-docstring)
foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
foo.py:4:0: E0244: Extending inherited Enum class "StringEnum" (invalid-enum-extension)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

pylint either checks the parent class for enum members or does enable this warning by default

### Pylint version

```shell
pylint 2.14.1
astroid 2.11.5
Python 3.8.5
```

### OS / Environment

Windows

**Comment by DanielNoord** [comment]
I can't reproduce this locally:
```console
❯ cat bar.py
from enum import Enum


class StringEnum(str, Enum):
    def __repr__(self) -> str:
        return f"<{self.__class__.__name__}.{self.name}>"
❯ cat foo.py
from bar import StringEnum


class ChatType(StringEnum):

    SENDER = "sender"
    PRIVATE = "private"
❯ pylint --version
pylint 2.14.1
astroid 2.11.5
Python 3.10.0 (default, Oct 27 2021, 16:20:25) [Clang 13.0.0 (clang-1300.0.29.3)]
❯ pylint --enable=invalid-enum-extension foo.py
************* Module foo
foo.py:1:0: C0114: Missing module docstring (missing-module-docstring)
foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
foo.py:4:0: C0115: Missing class docstring (missing-class-docstring)

------------------------------------------------------------------
Your code has been rated at 2.50/10 (previous run: 2.50/10, +0.00)
```

**Comment by Bibo-Joshi** [comment]
Thanks for the reply! I did some more testing and apparently I missed something while putting together the first MWE. My apologies for that. I now think that a `__slots__` entry in the parent is the problem, maybebecause it's interpreted as enum member.

New MWE:

```python
> cat .\foo.py
"""module"""
from enum import Enum


class StringEnum(str, Enum):
    __slots__ = ("a", "b")


class ChatType(StringEnum):

    SENDER = "sender"
    PRIVATE = "private"
> pylint --version
pylint 2.14.1
astroid 2.11.5
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
> pylint foo.py
************* Module foo
foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
foo.py:5:0: C0115: Missing class docstring (missing-class-docstring)
foo.py:9:0: C0115: Missing class docstring (missing-class-docstring)
foo.py:9:0: E0244: Extending inherited Enum class "StringEnum" (invalid-enum-extension)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

**Comment by Bibo-Joshi** [comment]
The discussion in #7257 lead to https://github.com/PyCQA/astroid/issues/1730. I leave it up to the maintainers to decide whether to keep this issue open.

**Comment by PabloAlexis611** [comment]
Hello there!

I also experience this issue when using [`django-enumfield`](https://pypi.org/project/django-enumfield/).

For example:

``` python

from django_enumfield import enum

class Animals(enum.Enum):
  CAT = 1
  DOG = 2
  SHARK = 3
  
  __labels__ = {
    CAT: "Cat",
    DOG: "Dog",
    SHARK: "SHARK!!!",
  }
```

I then get the following pylint error: _pylint: invalid-enum-extension / Extending inherited Enum class "Enum"_

**Comment by tales-aparecida** [comment]
Hi! Thanks for the great tool, and sorry for bumping this. Let me know if I should report this in astroid instead ;)

### False positive `invalid-enum-extension` with `enum.nonmember` in base class

`enum.nonmember()` assignments in an Enum base class are incorrectly treated as enum members by astroid, causing pylint to emit E0244 when the base is extended in another module.

This is a new instance of #6887, triggered by Django 6.0 dropping its `if PY311:` guard around `enum.nonmember()` usage in `models.Choices`.

#### Reproducer

Two files, declaring `BaseEnum` with `enum.nonmember` and `MyEnum(BaseEnum)`, then triggering `pylint` and printing `MyEnum.__members__`

```bash
podman run --rm docker.io/python:3.14-slim sh -c '
pip install -q pylint==4.0.6 2>&1 | tail -1
mkdir /tmp/repro

cat > /tmp/repro/base.py << "EOF"
import enum

class BaseEnum(enum.Enum):
    sentinel = enum.nonmember(True)
EOF

cat > /tmp/repro/child.py << "EOF"
from base import BaseEnum

class MyEnum(BaseEnum):
    FOO = "FOO"
EOF

echo "--- pylint output ---"
pylint --disable=all --enable=invalid-enum-extension /tmp/repro/child.py

echo "--- runtime check ---"
PYTHONPATH=/tmp/repro python3 -c "from child import MyEnum; print(MyEnum.FOO, dict(MyEnum.__members__))"
'
```

```
--- pylint output ---
child.py:3:0: E0244: Extending inherited Enum class "BaseEnum" (invalid-enum-extension)
--- runtime check ---
MyEnum.FOO {'FOO': <MyEnum.FOO: 'FOO'>}
```

The code is valid: `enum.nonmember()` excludes `sentinel` from `__members__` at runtime, so `BaseEnum` has no members and is safe to extend.

#### Root cause

astroid does not recognize `enum.nonmember()`. When it synthesizes `__members__` for `BaseEnum`, it includes the nonmember assignment as if it were a regular enum member:

```python
# astroid's synthesized locals for BaseEnum
>>> cls.locals["__members__"]
[Dict(items=[("'sentinel'", 'sentinel')])]  # should be empty
```

```bash
# Reproducer for the astroid synthesized locals
podman run --rm docker.io/python:3.14-slim sh -c '
pip install -q pylint==4.0.6 2>&1 | tail -1

cat > /tmp/repro.py << "EOF"
import enum

class BaseEnum(enum.Enum):
    sentinel = enum.nonmember(True)
EOF

python3 -c "
import astroid
mod = astroid.parse(open(\"/tmp/repro.py\").read())
cls = mod.body[1]
members = cls.locals[\"__members__\"]
for m in members:
    print([(k.as_string(), v.as_string()) for k, v in m.items])
"
'
```

`_check_enum_base` then reads this incorrect `__members__` and fires E0244.

#### How I hit this

Django 5 wrapped the `nonmember()` call in `if PY311:`, which astroid doesn't evaluate statically, so it never saw the assignment. Django 6 dropped that guard (minimum Python is now 3.11+), making the assignment unconditional and visible to astroid.

```bash
# Django 5: no error
podman run --rm docker.io/python:3.14-slim sh -c '
pip install -q "django>=5,<6" pylint==4.0.6 2>&1 | tail -1
cat > /tmp/repro.py << "EOF"
from django.db import models

class MyChoices(models.Choices):
    FOO = "FOO", "A foo"
EOF
pylint --disable=all --enable=invalid-enum-extension /tmp/repro.py
'

# Django 6: false positive
podman run --rm docker.io/python:3.14-slim sh -c '
pip install -q "django>=6,<7" pylint==4.0.6 2>&1 | tail -1
cat > /tmp/repro.py << "EOF"
from django.db import models

class MyChoices(models.Choices):
    FOO = "FOO", "A foo"
EOF
pylint --disable=all --enable=invalid-enum-extension /tmp/repro.py
'
```

```
# Django 5 output:
Your code has been rated at 10.00/10

# Django 6 output:
/tmp/repro.py:3:0: E0244: Extending inherited Enum class "Choices" (invalid-enum-extension)
```

#### Environment

- Python 3.14
- pylint 4.0.5, 4.0.6 (both affected)
- astroid 4.0.4


#### AI disclaimer

Assisted by Claude Opus 4.6, but I did try all commands I'm reporting here and double checked all info is accurate.

Kind regards,  
https://cki-project.org/
